index.d.ts

```diff
@@ -1,8 +1,8 @@
 export namespace OpenApi {
   export interface ApiOptions {
-      requestMiddleware?: (() => unknown) | undefined;
-      responseMiddleware?: (() => unknown) | undefined;
-      errorMiddleware?: (() => unknown) | undefined;
+      requestMiddleware?: ((event: object, context: object) => Promise<unknown>) | undefined;
```

Comment:
These don't have to be asysc methods, so it would be great if we could also include the `func<unknown>` as well. That prevents this from becoming a breaking change.

Reply:
That's a great idea, thanks! I added the old models back